examples/jsm/objects/SkyGPU.js

```diff
@@ -0,0 +1,190 @@
+import {
+	BackSide,
+	BoxGeometry,
+	Mesh,
+	NodeMaterial,
+	Vector3
+} from 'three';
+import { float, tslFn, vec3, acos, add, mul, clamp, cos, dot, exp, max, mix, modelViewProjection, normalize, positionWorld, pow, smoothstep, sub, varying, 	varyingProperty, vec4, uniform } from 'three/tsl';
+
+/**
+ * Based on "A Practical Analytic Model for Daylight"
+ * aka The Preetham Model, the de facto standard analytic skydome model
+ * https://www.researchgate.net/publication/220720443_A_Practical_Analytic_Model_for_Daylight
+ *
+ * First implemented by Simon Wallner
+ * http://simonwallner.at/project/atmospheric-scattering/
+ *
+ * Improved by Martin Upitis
+ * http://blenderartists.org/forum/showthread.php?245954-preethams-sky-impementation-HDR
+ *
+ * Three.js integration by zz85 http://twitter.com/blurspline
+*/
+
+class Sky extends Mesh {
+
+	constructor() {
+
+		const material = new NodeMaterial();
+
+		super( new BoxGeometry( 1, 1, 1 ), material );
+
+		this.turbidity = uniform( 2 );
+		this.rayleigh = uniform( 1 );
+		this.mieCoefficient = uniform( 0.005 );
+		this.mieDirectionalG = uniform( 0.8 );
+		this.sunPosition = uniform( new Vector3() );
+		this.up = uniform( new Vector3( 0, 1, 0 ) );
```

Comment:
@Mugen87 `Object3D` already defines an `up` property that is a normal `Vector3`. Thoughts on renaming this to something else to avoid any potential issues?

Reply:
Do you think `upUniform` would do the trick?

Reply:
Yeah, that should do it!

Reply:
https://github.com/mrdoob/three.js/pull/29199